### PR TITLE
PR from Sara: Ready for v1!

### PR DESCRIPTION
**Pull request by bmeyers**
- OSQP support in SD
- MOSEK continued support

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -28,6 +32,9 @@ jobs:
               sudo $pythonLocation/bin/python3 -m pip install awscli coverage
               sudo $pythonLocation/bin/python3 -m pip install setuptools --upgrade
               $pythonLocation/bin/python3 -m pip list
+              sudo git clone https://github.com/cvxgrp/signal-decomposition.git
+              sudo $pythonLocation/bin/python3 -m pip install -r signal-decomposition/requirements.txt
+              sudo $pythonLocation/bin/python3 -m pip install -e ./signal-decomposition/.
```
Comment (bmeyers) [inline .github/workflows/test.yml:37]: fix this once signal decomposition is pip installable from PyPI

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -28,6 +32,9 @@ jobs:
               sudo $pythonLocation/bin/python3 -m pip install awscli coverage
               sudo $pythonLocation/bin/python3 -m pip install setuptools --upgrade
               $pythonLocation/bin/python3 -m pip list
+              sudo git clone https://github.com/cvxgrp/signal-decomposition.git
```
Comment (bmeyers) [inline .github/workflows/test.yml:35]: point to PyPI instead of github

File: solardatatools/algorithms/clipping.py [hunk 3]
```diff
@@ -370,8 +370,8 @@ def calculate_cdf(self, data):
         y_rs = f(x_rs)
         return x_rs, y_rs
 
-    def make_problem(self, y, weight=1e1):
-        out = make_l2_l1d2_constrained(y, weight=weight)
+    def make_problem(self, y, weight=5, solver=None):
+        out = make_l2_l1d2_constrained(y, w1=weight, solver=solver)
```
Comment (bmeyers) [inline solardatatools/algorithms/clipping.py:374]: if it truly is not too much of a hassle, let's make this function align with the rest of the SD subroutines, i.e., return a result not a problem instance

File: solardatatools/algorithms/time_shifts.py [hunk 4]
```diff
@@ -61,46 +63,49 @@ def run(
...
             use_ixs = np.logical_and(use_ixs, ~np.isnan(metric))
         self.use_ixs = use_ixs
-        # Optimize c1
-        if c1 is None:
-            c1s = np.logspace(-1, 2, 11)
-            hn, rn, tv_metric, jpy, best_ix = self.optimize_c1(
-                metric, c1s, use_ixs, c2, periodic_detector, solver=solver
+        # Optimize w1
+        if w1 is None:
+            if solver == "MOSEK":
```
Comment (bmeyers) [inline solardatatools/algorithms/time_shifts.py:68]: add a "TODO" in the code to investigate in the future and possibly simplify

File: solardatatools/clear_day_detection.py [hunk 5]
```diff
@@ -25,7 +25,7 @@ def filter_for_sparsity(
             self,
             data,
             c1=6e3,
```
Comment (bmeyers) [inline solardatatools/clear_day_detection.py:27]: c1 --> w1(?)

File: solardatatools/clear_day_detection.py [hunk 6]
```diff
@@ -83,7 +85,7 @@ def find_clear_days(
         self.de = de/np.nanmax(de)
         # Solve a convex minimization problem to roughly fit the local 90th
         # percentile of the data (quantile regression)
-        self.x = tl1_l2d2p365(self.de, tau=0.9, c1=204697, yearly_periodic=False, solver=solver)
+        self.x = tl1_l2d2p365(self.de, tau=0.9, w1=204697, yearly_periodic=False, solver=solver)
```
Comment (bmeyers) [inline solardatatools/clear_day_detection.py:88]: can we make w1 one of 2e5 or 2.1e5?

**Review by bmeyers — COMMENTED** [review]
This looks great. Once the test builds complete successfully, let's merge with master.

**Review by pluflou — APPROVED** [review]

**Pull request merged**